src/components/Header/Header.js

```diff
@@ -60,13 +75,48 @@ export class Header extends PureComponent {
               />
             </Link>
             <div className="header__delivery-info">
-              <Input
-                name="address"
-                value={address}
-                placeholder="Address"
-                iconUrl="./images/locationImg.svg"
-                onChange={this.handleChange}
-              />
+              <div className="header__locations-info">
+                <Input
+                  name="address"
+                  value={address}
+                  placeholder={addressPlaceholder || 'Address'}
+                  iconUrl="./images/locationImg.svg"
+                  onChange={this.handleChange}
+                />
+
+                {
+                  this.state.address && (
+                    <div className="header__dropdown" id="header__dropdown">
+                      {
+                        locationsVariants
+                          .filter(({ title }) => title
+                            .toLowerCase()
+                            .includes(this.state.address.toLowerCase()))
+                          .map(({ title, id }) => (
+                            <button
+                              className="header__dropdown-button"
+                              type="button"
+                              key={id}
+                              onClick={() => this
+                                .handleDropdownVariantsClick(id, title)}
+                            >
+                              <Link to="/">
+                                <img
+                                  src="./images/locationImg.svg"
+                                  alt="search icon"
+                                  className="header__dropdown-button-img"
+                                />
+                                <p className="header__dropdown-button-title">
+                                  {title}
+                                </p>
+                              </Link>
+                            </button>
+                          ))
+                      }
+                    </div>
+                  )
+                }
```

Comment:
Should be extracted to separate component Autocomplete. Maybe with Input, maybe without. Think about component API which will be convenient to use elsewhere.